### PR TITLE
[RemoveDIs][NFC] Fix rotten green C API test

### PR DESCRIPTION
**Pull request by OCHyams**
`llvm_test_dibuilder(/*NewDebugInfoMode=*/true)` isn't currently executed in 
`return llvm_test_dibuilder(false) && llvm_test_dibuilder(true);`
because `llvm_test_dibuilder` returns 0 for success.

Split the llvm-c-test flag `--test-dibuilder` into two, one for the old and
one for the new debug informat. Add another lit test for the new mode.

Now that the test actually runs, it crashes using the new format with
`llvm/lib/IR/LLVMContextImpl.cpp:53:llvm::LLVMContextImpl::~LLVMContextImpl(): Assertion 'TrailingDbgRecords.empty() && "DbgRecords in blocks not cleaned"' failed. Aborted`.

Insert terminators into the blocks so that we don't leave the debug records
trailing, unattached to any instructions, which fixes that.

### CODE REVIEW

**Review by CarlosAlbertoEnciso — COMMENTED** [review]
General comment: in some places you use `mode` but in others you use `format`.
I would suggest to have consistent terms and options, based on the `format` term:

`new debuginfo format`.
`--test-dibuilder-debuginfo-new-format`
`--test-dibuilder-debuginfo-old-format`
`debug_info_new_format.ll`

File: llvm/tools/llvm-c-test/debuginfo.c [hunk 1]
```diff
@@ -215,10 +215,26 @@ int llvm_test_dibuilder(bool NewDebugInfoFormat) {
 
   LLVMDIBuilderFinalize(DIB);
 
+  // In the new debug mode, debug records get attached to instructions.
```
Comment (CarlosAlbertoEnciso) [inline llvm/tools/llvm-c-test/debuginfo.c:218]: Is this code depending on the given `NewDebugInfoFormat`?

Comment (OCHyams) [reply llvm/tools/llvm-c-test/debuginfo.c]: I thought it'd be better to minimise the difference between the tests. Adding the instructions into the old-format test doesn't hurt it IMO. In a future patch I will possibly be adding some `NewDebugInfoFormat`-specific behaviour, but I think minimising the difference is still worthwhile. What do you think?

**Review by jryans — APPROVED** [review]
Thanks, looks good to me! 😄

**Review by CarlosAlbertoEnciso — APPROVED** [review]
LGTM

**Comment by OCHyams** [comment]
Thanks both

**Pull request merged**